tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/upgradeTest/UpgradeCurrentVersion.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <br>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <br>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.tests.smoke.upgradeTest;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import java.io.File;
+
+import org.apache.activemq.artemis.tests.smoke.common.SmokeTestBase;
+import org.apache.activemq.artemis.tests.util.CFUtil;
+import org.apache.activemq.artemis.util.ServerUtil;
+import org.apache.activemq.artemis.utils.RandomUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/** This test is making sure the upgrade command would be able to upgrade a test I created with artemis 2.25.0 */
+public class UpgradeCurrentVersion extends SmokeTestBase {
+
+   File upgradedServer;
+
+   Process processServer;
+
+   @Before
+   public void beforeTest() throws Exception {
+      upgradedServer = new File(basedir + "/target/upgrade/currentVersion");
+
+      processServer = ServerUtil.startServer(upgradedServer.getAbsolutePath(), "upgradedServer", 0, 5000);
+      addProcess(processServer);
+   }
+
+   @Test
+   public void testSimpleSendReceive() throws Throwable {
```

Comment:
I think a likely better and faster test would be to comparable to / part-of the existing CompareUpgradeTest. We already have a current-expected-instance-config for those existing tests to compare with their old-upgraded-instance output to check for match. We can compare a new current-version-upgraded instance config with that and check they match in much the same fashion, whereas this test wouldnt notice if it did change something unless that actually breaks the server.

Reply:
though the other test uses a split-config and so one would need a rework to use the same comparison config for both.

Reply:
@gemmellr I just want this test to validate if the server works after a self upgrade. (Which could be used to change home... etc...)